packages/ketcher-react/src/script/editor/tool/AbstractTool.ts

```diff
@@ -0,0 +1,35 @@
+import Editor from '../Editor'
+
+interface ToolEventHandler {
+  click?(event: Event): void
+
+  dblclick?(event: Event): void
+
+  mousedown?(event: Event): void
+
+  mousemove?(event: Event): void
+
+  mouseup?(event: Event): void
+
+  mouseleave?(event: Event): void
+
+  mouseLeaveClientArea?(event: Event): void
+
+  mouseover?(event: Event): void
+}
+
+export interface AbstractTool extends ToolEventHandler {
```

Comment:
I thought TypeScript had a keyword [abstract](https://www.typescriptlang.org/docs/handbook/2/classes.html#abstract-classes-and-members) which was specific for abstract class. 

I'm not sure if we can:
```suggestion
abstract class ToolEventHandler {}
abstract class Tool extends ToolEventHandler {}
```

I thought the advantage of `interface` was there would be no extra JavaScript output, but one good thing about `abstract` was if we forget to implement the base class’s abstract members, we’ll get an error.

What's your take on this?

Reply:
I doubt using the base class as it often tends to use real inheritance with class extends and common logic, which is often tough to manage.
I agree that the 'Abstract' word could lead to misunderstanding. 
So my proposal is to rename `AbstractTool` to `Tool` and use it as is as a common interface.

@yuleicul let me know if you have other ideas